src/hotspot/share/runtime/sweeper.cpp

```diff
@@ -266,15 +266,14 @@ void NMethodSweeper::force_sweep() {
  */
 void NMethodSweeper::handle_safepoint_request() {
   JavaThread* thread = JavaThread::current();
-  if (SafepointMechanism::should_process(thread)) {
-    if (PrintMethodFlushing && Verbose) {
-      tty->print_cr("### Sweep at %d out of %d, yielding to safepoint", _seen, CodeCache::nmethod_count());
-    }
-    MutexUnlocker mu(CodeCache_lock, Mutex::_no_safepoint_check_flag);
-
-    ThreadBlockInVM tbivm(thread);
-    thread->java_suspend_self();
```

Comment:
AFAICS the only change needed in this function was to delete the java_suspend_self as it is not handled in the TBIVM.

Reply:
Fixed